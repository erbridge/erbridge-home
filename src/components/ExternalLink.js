import PropTypes from 'prop-types';
import React, { Component } from 'react';

class ExternalLink extends Component {
  static propTypes = {
    to: PropTypes.string.isRequired,
    className: PropTypes.string,
    children: PropTypes.node,
  };

  render() {
    const { children, className, to } = this.props;

    return (
      <a
        className={className}
        href={to}
        target="_blank"
        rel="noopener noreferrer"
      >
        {children}
      </a>
    );
  }
}

export default ExternalLink;
