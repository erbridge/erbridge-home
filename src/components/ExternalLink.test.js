import { shallow } from 'enzyme';
import React from 'react';

import ExternalLink from './ExternalLink';

it('renders correctly', () => {
  expect(
    shallow(
      <ExternalLink className="Test__link" to="http://example.com/">
        Example
      </ExternalLink>,
    ),
  ).toMatchSnapshot();
});
