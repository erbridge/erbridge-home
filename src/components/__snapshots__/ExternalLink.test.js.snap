// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="Test__link"
  href="http://example.com/"
  rel="noopener noreferrer"
  target="_blank"
>
  Example
</a>
`;
